client/extensions/woocommerce/app/order/order-details.js

```diff
@@ -66,32 +119,34 @@ class OrderDetails extends Component {
 					<div className="order__details-totals">
 						<div className="order__details-total-discount">
 							<div className="order__details-totals-label">{ translate( 'Discount' ) }</div>
-							<div className="order__details-totals-value">{ order.discount_total }</div>
+							<div className="order__details-totals-value">
+								{ formatCurrency( order.discount_total, order.currency ) || order.discount_total }
+							</div>
 						</div>
 						<div className="order__details-total-shipping">
 							<div className="order__details-totals-label">{ translate( 'Shipping' ) }</div>
-							<div className="order__details-totals-value">{ order.shipping_total }</div>
+							<div className="order__details-totals-value">
+								{ formatCurrency( order.shipping_total, order.currency ) || order.shipping_total }
+							</div>
 						</div>
 						<div className="order__details-total">
 							<div className="order__details-totals-label">{ translate( 'Total' ) }</div>
-							<div className="order__details-totals-value">{ order.total }</div>
+							<div className="order__details-totals-value">
+								{ formatCurrency( order.total, order.currency ) || order.total }
+							</div>
 						</div>
+						{ refundValue
+							? <div className="order__details-total-refund">
+								<div className="order__details-totals-label">{ translate( 'Refunded' ) }</div>
```

Comment:
Could we maybe move this out to a separate function and call it here? The large ternary feels weird.